source/geometry_model/box.cc

```diff
@@ -268,6 +269,115 @@ namespace aspect
       return std::min (std::max (d, 0.), maximal_depth());
     }
 
+    template <int dim>
+    void
+    Box<dim>::
+    update ()
+    {
+      this->get_pcout() << "   Updating surface values... " << std::endl;
+      TimerOutput::Scope timer_section(this->get_computing_timer(), "Geometry model surface update");
+
+      // loop over all of the surface cells and save the elevation to a stored value.
```

Comment:
Currently you do this computation unconditionally whenever the `update` function is called. However, that is expensive and has a number of problems:
- mesh deformation may not actually be active, then all of this is wasted
- in your current implementation `update` is called twice, once from [here](https://github.com/geodynamics/aspect/blob/e9cd1c8fc83f0d5c6e52c9d775a71382a1597049/source/simulator/core.cc#L645) and once from your new place [here](https://github.com/geodynamics/aspect/pull/5088/files#diff-72347c8b280f31adfdf4a3c112dbc352e9de7677d261e62dba25bef499fe674aR1897), only the second call is a useful update to the surface function.

As a first step you could only do this whole calculation if `this->get_parameters().mesh_deformation_enabled` is true. But then it would still be done twice per timestep. Additionally you could introduce a class member variable `bool need_surface_function_update`, which you set to false at the end of the update function. You then need to set this variable to true whenever the mesh changes, which you can do by connecting a small lambda function to the signal `this->simulator_signals().post_mesh_deformation`. This could look similar to the following (put this into the `initialize` function):
```
          this->get_signals().post_mesh_deformation.connect(
            [&](const SimulatorAccess<dim> &/*simulator_access*/)
          {
            need_surface_function_update = true;
          }
```

Reply:
Done.